e2e/app.spec.ts

```diff
@@ -62,58 +61,75 @@ describe('App', function() {
+   * 4. Navigating to the Actions List page by clicking on the adequate button on the Map page.
+   * 5. Scrolling to the bottom of the actions list page and trigger a new load.
+   * 6. Clicking on the first action on the actions list page, and thus displaying the correct Action page.
+   * 7. Clicking on the theme on the action page, and thus displaying the correct Theme page.
+   * 8. Clicking on the back button and thus returning to the previous Action page.
    */
   it('should allow a user to execute the main scenario', async function() {
-    this.timeout(15000);
+    this.timeout(20000);
+
+    /**
+     * 1. Arriving on the root page, which should be the Map Page.
+     */
 
     // Navigate to the map page.
     await mapPage.navigateTo();
     await expect(mapPage.getTitle()).to.eventually.equal('BioPocket');
 
-    // Wait for the actions list page to show up on the DOM
+    // Wait for the map page to show up on the DOM
     const mapPageFinder = mapPage.getPage();
     await presenceOf(mapPageFinder);
-    await expectDisplayed(mapPageFinder, 'MapPage is not displayed while it should be.');
+    await expectDisplayed(mapPageFinder, { elementName: 'Map Page' });
     await expect(mapPage.getPageTitle().getText()).to.eventually.equal('Carte');
 
     // Ensure that all markers are displayed.
     const markerIconFinders = await mapPage.getMarkerIcons();
     expect(markerIconFinders).to.have.lengthOf(3);
 
+    /**
+     * Clicking on a Location map marker, and thus displaying the popover with the correct information.
+     */
+
     // Click on the first marker (they are also sorted by ascending longitude and latitude).
-    markerIconFinders[0].click();
+    markerIconFinders[ 0 ].click();
 
     // Ensure that the popover is displayed.
     const popoverFinder = mapPage.getPopover();
     await visibilityOf(popoverFinder);
-    await expectDisplayed(popoverFinder, 'Popover is not displayed while it should be');
+    await expectDisplayed(popoverFinder, { elementName: 'Popover' });
 
     // Ensure that the data of the correct location is displayed in the popover.
     // Since both the location fixtures and marker icons have been sorted by ascending longitude and
     // latitude, the first location should correspond to the first marker icon.
-    const location = locations[0];
+    const location = locations[ 0 ];
     const locationDetailsText = await mapPage.getLocationDetails().getText();
     expect(locationDetailsText).to.have.string(location.get('name'));
     expect(locationDetailsText).to.have.string(location.get('short_name'));
 
+    /**
+     * 3. Dismissing the popover by clicking on its backdrop.
+     */
+
     // To dismiss the popover, get its backdrop element.
     const popoverBackdropFinder = await mapPage.getPopoverBackdrop();
     await expect(popoverBackdropFinder.isPresent(), 'Popover backdrop is not present while it should be.').to.eventually.equal(true);
 
     popoverBackdropFinder.click();
```

Comment:
Similarly to my comment in a previous PR, this should be encapsulated into a method in `MapPageObject` to avoid manipulating the DOM directly in the test.